File: src/problems/leetcode-cn/3379. 转换数组/solution/java/Solution.java
class Solution {
    public int[] constructTransformedArray(int[] nums) {
        int l = nums.length;
        int[] result = new int[l];
        for (int i = 0; i < l; i++)
            result[i] = nums[((i + nums[i]) % l + l) % l];
        return result;
    }
}

File: src/problems/leetcode-cn/3379. 转换数组/solution/rust/solution.rs
impl Solution {
    pub fn construct_transformed_array(nums: Vec<i32>) -> Vec<i32> {
        let (n, mut result) = (nums.len() as i32, vec![0; nums.len()]);
        for i in 0..nums.len() {
            result[i] = nums[(((((i as i32)+nums[i])%n)+n)%n) as usize];
        }
        result
    }
}